x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/RuntimeFieldMapper.java

```diff
@@ -75,8 +77,8 @@ protected void parseCreateField(ParseContext context) {
     }
 
     @Override
-    public ValueFetcher valueFetcher(MapperService mapperService, String format) {
-        throw new UnsupportedOperationException();
+    public ValueFetcher valueFetcher(MapperService mapperService, SearchLookup lookup, String format) {
+        return new DocValueFetcher(fieldType().docValueFormat(format, null), lookup.doc().getForField(fieldType()));
```

Comment:
thinking out loud, would it make any sense to call fieldType().fieldDataBuilder directly instead of calling getForField against the lookup? Any substantial difference between the two?

Reply:
I already commented on this, but I wonder if this SearchLookup argument should rather be a supplier. What are the pros and cons?

Reply:
One more: will the time zone stay null or is it only temporary?

Reply:
Replied above about supplier. Its a good idea I think.

About calling the builder directly - I think we need the lookup so runtime fields can have it.

Reply:
And time zone will stay null until the `fields` parameter supports it. `docvalue_fields` never has so I don't expect `fields` will do it any time soon. But is certainly possible!

Reply:
> About calling the builder directly - I think we need the lookup so runtime fields can have it.

yea we would still need to provide the lookup as a supplier to fielddatabuilder, we would get to the same method, just in a different way, I think

Reply:
> docvalue_fields never has so I don't expect fields will do it any time soon. But is certainly possible!

Cool, I saw later that it was already null in the doc values fetch phase so nothing new, I thought it had to do with limitations of runtime fields but that is not the case